mevshare/sim_result_backend.go

```diff
@@ -56,38 +56,41 @@ func (s *SimulationResultBackend) SimulatedBundle(ctx context.Context,
 	}
 	logger := s.log.With(zap.String("bundle", hash.Hex()))
 
-	// failed bundle does not go to the builder
-	err := s.store.InsertBundleForStats(ctx, bundle, sim)
-	if err != nil {
-		if errors.Is(err, ErrKnownBundle) {
-			logger.Debug("Bundle already known", zap.Error(err))
-			return nil
+	go func(ctx context.Context) {
+		// failed bundle does not go to the builder
+		if !sim.Success {
+			return
 		}
-		logger.Error("Failed to insert bundle for stats", zap.Error(err))
-	}
-	if !sim.Success {
-		return nil
-	}
 
-	err = s.ProcessHints(ctx, bundle, sim)
-	if err != nil {
-		logger.Error("Failed to process hints", zap.Error(err))
-	}
+		err := s.ProcessHints(ctx, bundle, sim)
+		if err != nil {
+			logger.Error("Failed to process hints", zap.Error(err))
+		}
+
+		for _, builder := range s.builders {
+			err := builder.SendMatchedShareBundle(ctx, bundle)
+			if err != nil {
+				logger.Warn("Failed to send bundle to builder", zap.Error(err))
+			}
+		}
 
-	for _, builder := range s.builders {
-		err := builder.SendMatchedShareBundle(ctx, bundle)
+		err = s.store.InsertBundleForBuilder(ctx, bundle, sim)
 		if err != nil {
-			logger.Warn("Failed to send bundle to builder", zap.Error(err))
+			logger.Error("Failed to insert bundle for builder", zap.Error(err))
 		}
-	}
 
-	err = s.store.InsertBundleForBuilder(ctx, bundle, sim)
+		s.externalBuilders.SendBundle(ctx, logger, bundle)
+	}(context.Background())
```

Comment:
I think it's worth passing context with timeout for this goroutine

Reply:
will do it properly before merging but this should do for now